fuse_core/include/fuse_core/eigen.h

```diff
@@ -79,6 +80,21 @@ std::string to_string(const Eigen::DenseBase<Derived>& m, const int precision =
   return oss.str();
 }
 
+template <typename Derived>
+bool isSymmetric(const Eigen::DenseBase<Derived>& m,
+                 const typename Eigen::DenseBase<Derived>::RealScalar precision =
+                     Eigen::NumTraits<typename Eigen::DenseBase<Derived>::Scalar>::dummy_precision())
+{
+  return m.isApprox(m.transpose(), precision);
```

Comment:
I didn't want to change behaviour here, so I still use `isApprox`, but according to https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#ae8443357b808cd393be1b51974213f9c and my experience, when the matrix `m` is zero (or very close to zero), the `precision` doesn't "work" as we'd expect.

Maybe we should use `isMuchSmallerThan` or a `cwise` comparison for the off-diagonal elements here. :thinking:

Reply:
Something like `((m - m.transpose()).cwiseAbs().maxCoeff() < precision)`? That works for me. That's what I'm doing in the `eigen_gtest.h` file.
https://github.com/locusrobotics/fuse/blob/devel/fuse_core/include/fuse_core/eigen_gtest.h#L99

Reply:
Changed to the expression you've suggested.